dali/image/webp.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021, Lennart Behme. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/image/webp.h"
+#include "dali/core/byte_io.h"
+
+namespace dali {
+
+WebpImage::WebpImage(const uint8_t *encoded_buffer, size_t length, DALIImageType image_type) :
+         GenericImage(encoded_buffer, length, image_type) {
+}
+
+Image::Shape WebpImage::PeekShapeImpl(const uint8_t *encoded_buffer, size_t length) const {
+  DALI_ENFORCE(encoded_buffer);
+  DALI_ENFORCE(length >= 16);
+
+  // RIFF header has 12 bytes
+  const uint8_t *vp8_data = encoded_buffer + 12;
+  if (vp8_data[0] == 'V' && vp8_data[1] == 'P' && vp8_data[2] == '8' && vp8_data[3] == ' ') {
+    // Simple file format (lossy)
```

Comment:
As we are reading here two bytes after vp8_data+16, can you add `DALI_ENFORCE(length >= 30);` (12 + 16 + 2).

Reply:
Fixed in 727f9047f9185f49a6b05421af71fbfa7f45b3bf